src/rgw/rgw_crypt.cc

```diff
@@ -1214,11 +1343,13 @@ int rgw_s3_prepare_decrypt(struct req_state* s,
     res = get_actual_key_from_kms(s->cct, key_id, key_selector, actual_key);
     if (res != 0) {
       ldout(s->cct, 10) << "No encryption key for key-id=" << key_id << dendl;
+      s->err.message = key_id + " does not exist";
       return res;
     }
     if (actual_key.size() != AES_256_KEYSIZE) {
       ldout(s->cct, 0) << "ERROR: key obtained from key_id:" <<
           key_id << " is not 256 bit size" << dendl;
+      s->err.message = "Server Side Encryption with KMS managed key must provide a valid key id.";
```

Comment:
this is the decrypt side, where we use the kms-keyid stored with the object instead of one provided in request headers. i think this should say something like `KMS provided an invalid key for the given kms-keyid`

Reply:
Yeah, OK